xmppserver/src/main/webapp/muc-room-affiliations.jsp

```diff
@@ -115,6 +117,15 @@
                     Element item = frag.addElement("item");
                     item.addAttribute("affiliation", affiliation);
                     item.addAttribute("jid", memberJID);
+                    if(nickName != null){
+                        item.addAttribute("nick", nickName);
+                    }else{
+                        //set the name of the user as the nickname of the member
+                        if(userJID != null){
+                            String username = JID.escapeNode(userJID.substring(0, userJID.indexOf('@')));
```

Comment:
Instead of parsing the string directly, it might be wiser to create a JID instance using the string, and get it's node.

Reply:
@guusdk is this resolved?

Reply:
Yes.